frontend/database/helpers.ts

```diff
@@ -0,0 +1,94 @@
+import { lastOfArray, RxCollection } from "rxdb";
+import { replicateRxCollection, RxReplicationState } from "rxdb/plugins/replication";
+import { BehaviorSubject } from "rxjs";
+
+const pushing$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+const pulling$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+/**
+ * Create a synchtonization process between the FE database and the BE database.
+ * @param collection Collection to replicate
+ * @param replicationIdentifier ID name for the replica
+ * @param primaryKey Name of the property that will be the primary key
+ * @param pushFunc Handler function to execute when doing a push procress
+ * @param pullFunc Handler function to execute when doing a pull process
+ * @returns List of handy functions to control the replication process
+ */
+export const setupReplication: <T extends { updatedAt: number; deleted: boolean }, PK>(
+  collection: RxCollection<T> | null,
+  replicationIdentifier: string,
+  primaryKey: keyof T,
+  pushFunc: (items: T[]) => Promise<void>,
+  pullFunc: (minTimestamp: number, lastId: PK | null, batchSize: number) => Promise<T[]>,
+) => [RxReplicationState<any, any>, number, BehaviorSubject<boolean>, BehaviorSubject<boolean>] = <
+  T extends { updatedAt: number; deleted: boolean },
+  PK,
+>(
+  collection: RxCollection<T> | null,
+  replicationIdentifier: string,
+  primaryKey: keyof T,
+  pushFunc: (items: T[]) => Promise<void>,
+  pullFunc: (minTimestamp: number, lastId: PK | null, batchSize: number) => Promise<T[]>,
+) => {
+  const replicationState = replicateRxCollection({
+    collection,
+    replicationIdentifier,
+    retryTime: 5 * 1000,
+    autoStart: false,
+    deletedField: "deleted",
+    push: {
+      handler: async (docs: any): Promise<any> => {
+        pushing$.next(true);
+        const store: T[] = docs.map((x: any) => x.newDocumentState) as any;
+
+        try {
+          console.debug(replicationIdentifier + " replica push", store);
```

Comment:
Not really no, it will only log if browser is in debug mode.

But maybe in production anyone can set its browser as debug mode and see information in console. I'll just remove it.